experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -353,6 +354,171 @@ describe("Editing", () => {
 
 			expectJsonTree([tree, tree1, tree2], [{ foo: [{}, "b"] }]);
 		});
+
+		describe.skip("Exhaustive removal tests", () => {
+			// Toggle the constant below to run each scenario as a separate test.
+			// This is useful to debug a specific scenario but makes CI and the test browser slower.
+			// Note that if the numbers of nodes and peers are too high (more than 3 nodes and 3 peers),
+			// then the number of scenarios overwhelms the test browser.
+			// Should be committed with the constant set to false.
+			const individualTests = false;
+			const nbNodes = 3;
+			const nbPeers = 3;
+			const testRemoveRevive = true;
+			const testMoveReturn = true;
+			assert(testRemoveRevive || testMoveReturn, "No scenarios to run");
+
+			const [outerFixture, innerFixture] = individualTests
+				? [describe, it]
+				: [it, (title: string, fn: () => void) => fn()];
+
+			type ScenarioStep =
+				| { readonly removeIdx: number; readonly peer: number }
+				| { readonly peer: number };
```

Comment:
IMO worth writing a brief comment on what each step does--first one is obvious but second one less so (especially as the generator function with docs saying what tests do is below; I definitely wouldn't have guessed it meant "undo")

```suggestion
			type ScenarioStep =
				// Represents `peer` removing a node
				| { readonly removeIdx: number; readonly peer: number }
				// Represents `peer` undoing its most recent change
				| { readonly peer: number };
```